bsip-0087.md

```diff
@@ -0,0 +1,35 @@
+    BSIP: 0087
+    Title: Force Settlement Fee Ratio
+    Authors:
+      Jerry Liu bitcrab@qq.com
+    Status: Draft
+    Type: Protocol
+    Created: 2020-02-23
+    Discussion: https://github.com/bitshares/bsips/issues/260
+    Worker: TBD
+
+# Abstract
+This BSIP provide a solution to charge fee from force settlement.
+
+# Motivation
+Force settlement is an important part in smartcoin design, it provides the power to smartcoin holders to ask for collaterals with reference to feed price, and then strengthens the peg of smartcoin.
+
+The community continuously optimize the rules of smartcoin to balance the interests of relevant parties, taking bitCNY as an example, at the beginning the rules focus more on smart coin peg but care less on debt position owners’ interests, change are done to give debt position owners more protection, the force settlement offset has been changed from 0 to 2%, the max settlement  volume per maintenance period has been changed from no limit to 0.5% of smartcoin supply.  
+
+In the past several weeks, force settlement happened more intensively than before, the background is that the crypto market seemly switched from bear to bull, and BTS price went above 0.22CNY - which is the voted threshold per BSIP76, at some time the frequently happened force settlements made the lowest debt position collateral ratio even higher than 3, this phenomenon triggered warm discussion, almost all the users agree that the settler need to pay more cost while executing force settlement, and the cost will be paid to system as fee.
+
+# Rationale
+Force settlement fee can be seen as to pay part of the cost for smartcoin supply, stabilization and also the liquidation of collaterals, it is irrelevant to market fee sharing.
+To ensure the debt positions can always be closed with suffcient smartcoin, it is more feasible to cut off part of the traded collaterals as fee, instead of smartcoin.
+
+# Specification
+Add one new parameter Force Settlement Fee Percentage(FSFP) for each smartcoin, which is controlled by the smartcoin owner.
+
+FSO is Force Settlement Offset.
+
+When a force settlement is executed, the buyer sells smartcoin with quantity `X` and get collaterals in quantity `X*(1-FSFP)/[feed price price*(1+FSO)]`, the settled debt position owner sells collaterals in quantity `X/[feed price*(1+FSO)]` and get smartcoin in quantity `X`, the delta in paid and received collaterals in quantity `X*FSFP/[feed price*(1+FSO)]` will be paid to the owner of the smartcoin as force settlement fee.
```

Comment:
The current definition of FSFP is confusing imo.

FSO is a percent, and FSFP is a percent. For the sake of clarity of my explanation, let's have both be denominated as values in `[0,1]` where `1` is `100%`. I would suggest to refer them both to the same basis, namely the feed price.

Current feed price is `Z BTS/bitABC`. I want to settle `X bitABC`. Now the formula to calculate the collateral `Y BTS` I receive should follow `Y = X*Z*(1-FSFP-FSO)`, i.e. the percentages are both in reference to `X*Z`. 

Now to coin the wording: The force settlement order price is `Z*(1-FSFP-FSO)` (maker price).
Then the debt position holder sells collateral for `Z*(1-FSO)` (i.e. he gets to keep `Z*FSO` collateral as compensation), and the platform receives `Z*FSFP` as fee.

Reply:
@abitmore please advise, I feel the important thing is to keep it compatible to core code and BSIP74, which is already voted up.

Reply:
I still think it's a UI thing. `Y=X*Z*(1-FSFP)*(1-FSO)` is as clear as `Y=X*Z*(1-FSFP-FSO)`, and the former is even clearer when implementing in the core.

Reply:
With `Y=X*Z*(1-FSFP)*(1-FSO)` the two parameters are dependent.

e.g. `X=100 bitABC` `Z=10 BTS/bitABC` `FSFP=0.5` (50%) `FSO=0.1`. Then we have
`Y = 1000*0.5*0.9=450`. It is also not explicitly clear how the distribution works (How much does the network get, how much does debt position holder get?). I need to know in what order the offset/fee are applied. In your case you define it that first the offset is applied, and then the fee. In the above example the debt position holder would receive `100 BTS` and the platform `450 BTS`. I think that is counterintuitive, at least I assumed that the platform would get `500 BTS` and the owner `100 BTS`.

To make a drastic example, if `FSO=0.5` holds then `FSFP` only gets its percent of half the settled value. If you really intend this kind of dependency please highlight it in a clear way in a remark. Asset management is already highly complicated and this introduces and unnecessary additional complication (imho).

Reply:
either formula is acceptable in math, if you are not satisfied with 100/450, you can get 100/500 easily by setting a higher FSFP, here we need to put higher priority on the compatibility to core code.

so just let core team member decide, OK? I don't think we need to waste so much time on such a thing.

Reply:
> either formula is acceptable in math, if you are not satisfied with 100/450, you can get 100/500 easily by setting a higher FSFP, here we need to put higher priority on the compatibility to core code.
> 
> so just let core team member decide, OK? I don't think we need to waste so much time on such a thing.

Certainly, if its better for performance reason. 

The meaning of the parameters needs to be explained properly, especially if they have a dependence in the displayed formula. A BSIP is the reference for the common BTS holders to decide if they want to vote for it or not, and needs to be understandable.

> either formula is acceptable in math, if you are not satisfied with 100/450, you can get 100/500 easily by setting a higher FSFP

This is just an example highlighting that the parameters can be easily misunderstood if not explained properly.

So keeping your formula is fine, but please add an explanation of the dependency of the two parameters on each other.

Reply:
update #Specification.

actually in my draft I suppose the current `force settlement price = feed price*(1+FSO)`, as what is shown in GUI, however it seems in core code the `settlement order price = feed price/(1-FSO)`, example is that the current bitCNY force settlement price  shown at top of GUI is 0.2244, however the order price = 0.22449.

I feel it's better to let force settlement price = feed price*(1+FSO), however, seems if we follow the current core code, the specification should be modified as:

_When a force settlement is executed, the buyer sells smartcoin with quantity `X` and get collaterals in quantity `X*(1-FSO)*(1-FSFP)/feed price`, the settled debt position owner sells collaterals in quantity `X*(1-FSO)/feed price` and get smartcoin in quantity `X`, the delta in paid and received collaterals in quantity `X*FSFP*(1-FSO)/feed price` will be paid to the owner of the smartcoin as force settlement fee._

_In this scenario, `force settlement order price = feed price/(1-FSO)` is the price to the settled debt position owners(maker). and the force settlement fee is calculated out based on this order price and FSFP._

@abitmore please advise whether to do this modification.

Reply:
I think the modified text is clear. Thanks.

Reply:
> In summary, keeping your formula is fine, but please add an explanation of the dependency of the two parameters on each other.

I think now it's clearly explained in updated specification:

_In this scenario, as a price offset parameter, FSO is irrelevant to fee charging, it just define `force settlement price = feed price/(1-FSO)`. and the force settlement fee is paid in collateral and the amount is calculated out based on `force settlement price` and `FSFP`._